pyproject.toml

```diff
@@ -9,6 +9,8 @@ python = ">=3.9"
 mkdocs = "==1.5.3"
 mkdocs-material = "==9.5.3"
 mkdocs-awesome-pages-plugin = "==2.9.2"
+# to support amazonlinux 2 on vercel with python 3.9
+urllib3 = "<2"
```

Comment:
Is this new `urllib3` dependency really needed? I'm deploying a simple docs site with Vercel _without_ that dependency: see [my Git Gosling site on Vercel](https://git-gosling.vercel.app/)

Quote from [my `requirements.txt` file for Git Gosling](https://github.com/HonkingGoose/git-gosling/blob/main/requirements.txt):

> ```
> mkdocs-material==9.5.4
> mkdocs==1.5.3
> ```

Reply:
i get poetry errors otherwise:
```
poetry run mkdocs build
--
11:50:47.330 | Traceback (most recent call last):
11:50:47.330 | File "/vercel/.cache/pypoetry/virtualenvs/docs-renovatebot-com-HhybzZs7-py3.9/bin/mkdocs", line 8, in <module>
11:50:47.330 | sys.exit(cli())
11:50:47.330 | File "/vercel/.cache/pypoetry/virtualenvs/docs-renovatebot-com-HhybzZs7-py3.9/lib/python3.9/site-packages/click/core.py", line 1157, in __call__
11:50:47.330 | return self.main(*args, **kwargs)
11:50:47.331 | File "/vercel/.cache/pypoetry/virtualenvs/docs-renovatebot-com-HhybzZs7-py3.9/lib/python3.9/site-packages/click/core.py", line 1078, in main
11:50:47.331 | rv = self.invoke(ctx)
11:50:47.331 | File "/vercel/.cache/pypoetry/virtualenvs/docs-renovatebot-com-HhybzZs7-py3.9/lib/python3.9/site-packages/click/core.py", line 1688, in invoke
11:50:47.331 | return _process_result(sub_ctx.command.invoke(sub_ctx))
11:50:47.331 | File "/vercel/.cache/pypoetry/virtualenvs/docs-renovatebot-com-HhybzZs7-py3.9/lib/python3.9/site-packages/click/core.py", line 1434, in invoke
11:50:47.331 | return ctx.invoke(self.callback, **ctx.params)
11:50:47.331 | File "/vercel/.cache/pypoetry/virtualenvs/docs-renovatebot-com-HhybzZs7-py3.9/lib/python3.9/site-packages/click/core.py", line 783, in invoke
11:50:47.332 | return __callback(*args, **kwargs)
11:50:47.332 | File "/vercel/.cache/pypoetry/virtualenvs/docs-renovatebot-com-HhybzZs7-py3.9/lib/python3.9/site-packages/mkdocs/__main__.py", line 283, in build_command
11:50:47.332 | cfg = config.load_config(**kwargs)
11:50:47.332 | File "/vercel/.cache/pypoetry/virtualenvs/docs-renovatebot-com-HhybzZs7-py3.9/lib/python3.9/site-packages/mkdocs/config/base.py", line 378, in load_config
11:50:47.332 | errors, warnings = cfg.validate()
11:50:47.332 | File "/vercel/.cache/pypoetry/virtualenvs/docs-renovatebot-com-HhybzZs7-py3.9/lib/python3.9/site-packages/mkdocs/config/base.py", line 230, in validate
11:50:47.332 | run_failed, run_warnings = self._validate()
11:50:47.332 | File "/vercel/.cache/pypoetry/virtualenvs/docs-renovatebot-com-HhybzZs7-py3.9/lib/python3.9/site-packages/mkdocs/config/base.py", line 188, in _validate
11:50:47.332 | self[key] = config_option.validate(value)
11:50:47.332 | File "/vercel/.cache/pypoetry/virtualenvs/docs-renovatebot-com-HhybzZs7-py3.9/lib/python3.9/site-packages/mkdocs/config/config_options.py", line 182, in validate
11:50:47.333 | return self.run_validation(value)
11:50:47.333 | File "/vercel/.cache/pypoetry/virtualenvs/docs-renovatebot-com-HhybzZs7-py3.9/lib/python3.9/site-packages/mkdocs/config/config_options.py", line 1064, in run_validation
11:50:47.333 | self.load_plugin_with_namespace(name, cfg)
11:50:47.333 | File "/vercel/.cache/pypoetry/virtualenvs/docs-renovatebot-com-HhybzZs7-py3.9/lib/python3.9/site-packages/mkdocs/config/config_options.py", line 1102, in load_plugin_with_namespace
11:50:47.333 | return (name, self.load_plugin(name, config))
11:50:47.333 | File "/vercel/.cache/pypoetry/virtualenvs/docs-renovatebot-com-HhybzZs7-py3.9/lib/python3.9/site-packages/mkdocs/config/config_options.py", line 1120, in load_plugin
11:50:47.333 | plugin_cls = self.installed_plugins[name].load()
11:50:47.333 | File "/vercel/.cache/pypoetry/virtualenvs/docs-renovatebot-com-HhybzZs7-py3.9/lib/python3.9/site-packages/importlib_metadata/__init__.py", line 184, in load
11:50:47.334 | module = import_module(match.group('module'))
11:50:47.334 | File "/usr/local/lib/python3.9/importlib/__init__.py", line 127, in import_module
11:50:47.335 | return _bootstrap._gcd_import(name[level:], package, level)
11:50:47.335 | File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
11:50:47.335 | File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
11:50:47.335 | File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
11:50:47.335 | File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
11:50:47.335 | File "<frozen importlib._bootstrap_external>", line 850, in exec_module
11:50:47.335 | File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
11:50:47.335 | File "/vercel/.cache/pypoetry/virtualenvs/docs-renovatebot-com-HhybzZs7-py3.9/lib/python3.9/site-packages/material/plugins/privacy/plugin.py", line 28, in <module>
11:50:47.335 | import requests
11:50:47.335 | File "/vercel/.cache/pypoetry/virtualenvs/docs-renovatebot-com-HhybzZs7-py3.9/lib/python3.9/site-packages/requests/__init__.py", line 43, in <module>
11:50:47.335 | import urllib3
11:50:47.335 | File "/vercel/.cache/pypoetry/virtualenvs/docs-renovatebot-com-HhybzZs7-py3.9/lib/python3.9/site-packages/urllib3/__init__.py", line 41, in <module>
11:50:47.335 | raise ImportError(
11:50:47.335 | ImportError: urllib3 v2 only supports OpenSSL 1.1.1+, currently the 'ssl' module is compiled with 'OpenSSL 1.0.2k-fips  26 Jan 2017'. See: https://github.com/urllib3/urllib3/issues/2168
11:50:47.388 | make: *** [build-docs] Error 1
11:50:47.393 | Error: Command "make build" exited with 2
```
maybe pip uses urllib3 v1 on your case, because it's already installed globally

Reply:
I'm using pip on my project, you're using Poetry. Or maybe Vercel installs extra stuff for me in their build environment/container?

Reply:
no. the installed pip doesn't use urllib3 v2, so urllib3 v1 is already installed and mkdocs is satisfied by this.

I'm now installing a newer prebuild python with openssl v1.1 so it works with urllib3 v2+ now 💪